packages/use-contractkit/src/modals/connect.tsx

```diff
@@ -148,10 +149,11 @@ export const ConnectModal: React.FC<ConnectModalProps> = ({
           ([providerKey, provider]) =>
             typeof window !== 'undefined' &&
             provider.showInList() &&
-            Object.keys(allScreens).find((screen) => screen === providerKey)
+            providerKey in allScreens &&
+            providerKey.toLowerCase().includes(search.toLowerCase())
         )
         .sort(([, a], [, b]) => sort(a, b)),
-    [_providers, allScreens, sort]
+    [_providers, allScreens, sort, search]
```

Comment:
As it is now, the search being inside this `useMemo` leads to a bug where if you close the modal after a search, that search will be saved but the text in the input is not up to date. 
- Control search separately from `providers`
or 
- Control the value for the input using the search value (probably should be done regardless)
- Reset the `search` state on close
 
![Kapture 2022-04-28 at 11 17 28](https://user-images.githubusercontent.com/21973269/165720497-722271a6-6169-4b4a-a7bf-69c160f03286.gif)